spark/src/test/scala/io/delta/sharing/spark/DeltaSharingSourceCDFSuite.scala

```diff
@@ -355,6 +409,54 @@ class DeltaSharingSourceCDFSuite extends QueryTest
     }
   }
 
+  /**
+   * Test streaming checkpoint
+   */
+  integrationTest("CDF Stream - restart from checkpoint") {
+    withTempDirs { (checkpointDir, outputDir) =>
+      val processedRows = Seq(2, 3, 8, 2, 2)
+      val query = withStreamReaderAtVersion()
+        .option("maxFilesPerTrigger", "1")
+        .load().select("age").writeStream.format("parquet")
+        .option("checkpointLocation", checkpointDir.getCanonicalPath)
+        .start(outputDir.getCanonicalPath)
+      try {
+        query.processAllAvailable()
+        val progress = query.recentProgress.filter(_.numInputRows != 0)
+        assert(progress.length === processedRows.size)
+        progress.zipWithIndex.map { case (p, index) =>
+          assert(p.numInputRows === processedRows(index))
+        }
+      } finally {
+        query.stop()
+      }
+
+      // There are 5 checkpoints, remove the latest 3.
+      val checkpointFiles = FileUtils.listFiles(checkpointDir, null, true).asScala
```

Comment:
Consider adding a comment on why there are 5 checkpoints.
Can we assert on the size of checkpointFiles ?

Reply:
added.